src/expr.cpp

```diff
@@ -1576,11 +1579,11 @@ static llvm::Value *lEmitBinaryPointerArith(BinaryExpr::Op op, llvm::Value *valu
             }
 
             // And now do the actual division
-            return ctx->BinaryOperator(llvm::Instruction::SDiv, delta, size, "element_diff");
+            return ctx->BinaryOperator(llvm::Instruction::SDiv, delta, size, false, "element_diff");
         } else {
             // ptr - integer
             llvm::Value *zero = lLLVMConstantValue(type1, g->ctx, 0.);
-            llvm::Value *negOffset = ctx->BinaryOperator(llvm::Instruction::Sub, zero, value1, "negate");
+            llvm::Value *negOffset = ctx->BinaryOperator(llvm::Instruction::Sub, zero, value1, isSigned, "negate");
```

Comment:
This should be false, it's part of pointer arith logic. C++ is not adding `nsw` bit for it.